e2e/tests/upgrades/upgrade_test.go

```diff
@@ -386,6 +391,147 @@ func (s *UpgradeTestSuite) TestV5ToV6ChainUpgrade() {
 	})
 }
 
+func (s *UpgradeTestSuite) TestV6ToV7ChainUpgrade() {
+	t := s.T()
+	testCfg := testconfig.FromEnv()
+
+	ctx := context.Background()
+	relayer, channelA := s.SetupChainsRelayerAndChannel(ctx)
+	chainA, chainB := s.GetChains()
+
+	createClientOptions := ibc.CreateClientOptions{
+		TrustingPeriod: time.Duration(time.Hour * 24 * 7).String(),
+	}
+
+	// create second tendermint client
+	s.SetupClients(ctx, relayer, createClientOptions)
+
+	status, err := s.Status(ctx, chainA, ibctesting.FirstClientID)
+	s.Require().NoError(err)
+	s.Require().Equal("Active", status)
+
+	status, err = s.Status(ctx, chainA, ibctesting.SecondClientID)
+	s.Require().NoError(err)
+	s.Require().Equal("Active", status)
+
+	var (
+		chainADenom         = chainA.Config().Denom
+		chainBIBCToken      = testsuite.GetIBCToken(chainADenom, channelA.Counterparty.PortID, channelA.Counterparty.ChannelID) // IBC token sent to chainB
+		soloMachineClientID = "06-solomachine-2"
+	)
+
+	chainAWallet := s.CreateUserOnChainA(ctx, testvalues.StartingTokenAmount)
+	chainAAddress := chainAWallet.Bech32Address(chainA.Config().Bech32Prefix)
+
+	chainBWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
+	chainBAddress := chainBWallet.Bech32Address(chainB.Config().Bech32Prefix)
+
+	// create solo machine client
```

Comment:
I don't think this comment is very useful. Maybe we can outline **why** we're using the solomachine type from `ibctesting` instead. i.e. we don't have a real implementation so we can use this to simulate it.